src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2652,13 +2652,27 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             case NI_Sve2_BitwiseClearXor:
                 if (targetReg != op1Reg)
                 {
-                    assert(targetReg != op2Reg);
+                    assert(targetReg != op2Reg && targetReg != op3Reg);
                     GetEmitter()->emitInsSve_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, op1Reg);
                 }
                 // Always use the lane size D. It's a bitwise operation so this is fine for all integer vector types.
                 GetEmitter()->emitInsSve_R_R_R(ins, emitSize, targetReg, op2Reg, op3Reg, INS_OPTS_SCALABLE_D);
                 break;
 
+            case NI_Sve2_BitwiseSelect:
+            case NI_Sve2_BitwiseSelectLeftInverted:
+            case NI_Sve2_BitwiseSelectRightInverted:
+                // op1: select, op2: left, op3: right
+                // Operation is destructive on the 'left' operand.
+                if (targetReg != op2Reg)
```

Comment:
this should be `op1Reg` because 1st operand has RMW semantics, not 2nd.

Reply:
@kunalspathak this didn't need addressing because the logic was already correct, right?

The instruction is `BSL <Zdn>.D, <Zdn>.D, <Zm>.D, <Zk>.D` and the operation is:
```
CheckSVEEnabled();
constant integer VL = CurrentVL;
bits(VL) operand1 = Z[dn, VL];
bits(VL) operand2 = Z[m, VL];
bits(VL) operand3 = Z[k, VL];

Z[dn, VL] = (operand1 AND operand3) OR (operand2 AND NOT(operand3));
```

So we end up emitting `mov tgt, op2; BSL tgt, op3, op1`

Reply:
right, it was already resolved in https://github.com/dotnet/runtime/pull/115775#discussion_r2100464164